### PR TITLE
Shayan/Portal for setting modals

### PR DESCRIPTION
**Pull request by shayan-binary-2**

### CODE REVIEW

File: app/test.scss [hunk 1]
```diff
@@ -18,8 +18,23 @@
...
+            width: 100%;
+            box-sizing: content-box;
+        }
+        input[type='checkbox'] {
+            width: auto;
+        }
+        button {
+            border-radius: 4px;
+            background: #dedede;
+            border: 1px solid #aaa;
```
Comment (mahdiaryayi-fs) [inline app/test.scss:35]: Is it OK to use hex colors directly instead of Sass variables?

File: app/test.scss [hunk 2]
```diff
@@ -61,7 +76,12 @@
...
-.portal-node {
-    position: relative;
+.portal-node .cq-modal__overlay {
+    position: fixed;
     z-index: 9;
+    left: 0px;
+    top: 0px;
+    width: 100%;
+    height: 100%;
+    background: rgba(10, 10, 10, 0.3);
```
Comment (mahdiaryayi-fs) [inline app/test.scss:86]: same here

File: src/components/StudyLegend.jsx [hunk 3]
```diff
@@ -188,57 +188,61 @@ const StudyLegend = ({
...
+                        </p>
+                        <button
+                            type="button"
+                            className="sc-btn sc-btn--primary sc-btn--w100"
+                            onClick={() => onSelectItem(infoItem.id)}
+                        >
+                            {t.translate('Add')}
+                        </button>
+                    </div>
+                ) : ''}
```
Comment (mahdiaryayi-fs) [inline src/components/StudyLegend.jsx:229]: this ternary's `else` condition in an empty string. so can you use this `{infoItem && <div> ... </div>}` pattern instead and remove the ternary?

File: app/test.scss (hunk 1 above)
Comment (shayan-binary-2) [reply app/test.scss:35]: @mahdiaryayi-fs actually this is test file which we use it for developer test and no more. we use to break rules in this file from a long time ago ;)

File: src/components/StudyLegend.jsx (hunk 3 above)
Comment (shayan-binary-2) [reply src/components/StudyLegend.jsx:229]: @mahdiaryayi-fs agree I'll fix this issue

File: app/test.scss (hunk 2 above)
Comment (shayan-binary-2) [reply app/test.scss:86]: @mahdiaryayi-fs just like above

**Pull request merged**